batch_finder_resource_method.md

```diff
@@ -0,0 +1,310 @@
+    public BatchfindersBatchFindBySearchGreetingsBuilder criteriaParam(Iterable<GreetingCriteria> value) {
+        super.setReqParam("criteria", value, GreetingCriteria.class);
+        return this;
+    }
+
+    // the 2nd way to set the batch query parameter
+    public BatchfindersBatchFindBySearchGreetingsBuilder addCriteriaParam(GreetingCriteria value) {
+        super.addReqParam("criteria", value, GreetingCriteria.class);
+        return this;
+    }
+
+    // set common query parameter
+    public BatchfindersBatchFindBySearchGreetingsBuilder messageParam(java.lang.String value) {
+        super.setReqParam("message", value, java.lang.String.class);
+        return this;
+    }
+}
+```
+Here is an example to show how to use request builder to build BATCH_FINDER request.
+```java
+  @Test
+  public void testUsingResourceBuilder() throws RemoteInvocationException {
+    // define batch search criteria
+    GreetingCriteria c1 = new GreetingCriteria().setId(1L).setTone(Tone.SINCERE);
+    GreetingCriteria c2 = new GreetingCriteria().setId(2L).setTone(Tone.FRIENDLY);
+    GreetingCriteria c3 = new GreetingCriteria().setId(100);
+    
+    //set batch query parameter and common query parameter
+    Request<BatchCollectionResponse<Greeting>> req = new GreetingsRequestBuilders().batchFindBySearchGreetings()
+        .criteriaParam(Arrays.asList(c1, c2, c3)).messageParam("hello world").build();
+    Response<BatchCollectionResponse<Greeting>> resp = REST_CLIENT.sendRequest(req).getResponse();
+    BatchCollectionResponse<Greeting> response = resp.getEntity();
+  
+    List<BatchFinderCriteriaResult<Greeting>> batchResult = response.getResults();
+  
+    Assert.assertEquals(batchResult.size(), 3);
+  
+    // on success
+    List<Greeting> greetings1 = batchResult.get(0).getElements();
+    Assert.assertTrue(greetings1.get(0).hasTone());
+    Assert.assertTrue(greetings1.get(0).getTone().equals(Tone.SINCERE));
+  
+    // on error
+    ErrorResponse error = batchResult.get(2).getError();
+    Assert.assertTrue(batchResult.get(2).isError());
+    Assert.assertEquals(error.getMessage(), "Fail to find Greeting!");
+  }
+```
+
+
+## Resource API
+BATCH_FINDER is supported on Collection and Association Resources only(See more details about [Collection Resource](/rest.li/spec/protocol#collection-resources) and [Association Resource](/rest.li/spec/protocol#association-resources)).
+Resources may provide zero or more BATCH_FINDER resource methods. Each BATCH_FINDER method must be annotated with the @`BatchFinder` annotation.
+
+Pagination default to start=0 and count=10. Clients may set both of these parameters to any desired value.
+### Method Annotation and Parameters
+The @`BatchFinder` annotation takes 2 required parameter:
+- `value` : which indicates the BATCH_FINDER method name
+- `batchParam` : which indicates the name of the batch criteria parameter, each BATCH_FINDER method must have and can only have one batch parameter
+
```

Comment:
It might be useful to have an example of how to call this resource method with curl to highlight the mapping between the query parameter `bq` and the `value` parameter in the `BatchFinder` annotation.